File: src/python/oop/ElevatorSystem/elevator/hole_buttons_panel.py
from .elevator import *
from .elevator_status import *
from .direction import *

class HoleButtonsPanel:
    def __init__(self, floor: int, system: ElevatorStatus):
        self._floor = floor
        self._system = system

    def press_up(self) -> Elevator:
        return self._system.call_elevator(self._floor, Direction.UP)

    def press_down(self) -> Elevator:
        return self._system.call_elevator(self._floor, Direction.DOWN)

File: src/csharp/oop/ElevatorSystem/Elevator/HoleButtonsPanel.cs
namespace ElevatorSystem.Elevator;

public class HoleButtonsPanel(int floor, ElevatorSystem elevatorSystem)
{
    public ElevatorCabin? PressUp()
        => elevatorSystem.CallElevator(floor, Direction.Up);

    public ElevatorCabin? PressDown()
        => elevatorSystem.CallElevator(floor, Direction.Down);
}
